pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -444,28 +481,51 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
                         const std::vector<size_t> &wires, bool inverse,
                         ParamT angle) {
         assert(wires.size() == 1);
+        const size_t rev_wire = num_qubits - wires[0] - 1;
+        const size_t rev_wire_shift = (static_cast<size_t>(1U) << rev_wire);
+        const size_t wire_parity = fillTrailingOnes(rev_wire);
+        const size_t wire_parity_inv = fillLeadingOnes(rev_wire + 1);
```

Comment:
As these tend to exist in the same form for many other methods, should we offload them to a private function?

Reply:
Moved.